src/renderer/components/dock/logs.tsx

```diff
@@ -51,112 +59,178 @@ export class Logs extends React.Component<Props> {
   }
 
   componentDidMount() {
-    disposeOnUnmount(this,
-      reaction(() => this.props.tab.id, this.reload, { fireImmediately: true }),
-    );
+    disposeOnUnmount(this, [
+      kubeWatchApi.subscribeStores([
+        podsStore,
+      ], {
+        namespaces: logTabStore.getNamespaces(),
+      }),
+      when(() => this.canLoad, () => this.load(true)),
+      disposingReaction(
+        () => [this.tabId, this.tabData] as const,
+        ([curTabId], [oldTabId]) => {
+          if (curTabId !== oldTabId) {
+            logStore.clearLogs(this.tabId);
+          }
+
+          return when(() => this.canLoad, () => this.load(true));
+        },
+        {
+          equals: comparer.structural,
+        },
+      ),
+      disposingReaction(() => this.tabData, (data) => {
```

Comment:
Can it be done with standard mobx reactions? Or event with good old `componentDidUpdate` props comparisons? Just thinking if we decided to use reactions less often and now we have one more reaction method. However, I have nothing against it to have one, maybe just need to add tests and first of all weight its necessity?

Reply:
For the specific call you commented on here I have replaced it with a plain `reaction`. However, with the call above I think we actually need it. Because there is an inherent race between switching of tabs or pods or containers and being able to load new logs.